### PR TITLE
Qsim upgrade

### PR DESCRIPTION
**Pull request by MichaelBroughton**
cc: @95-orion-martin , @we-taper

Finishes step 1 in #422 

Thematically speaking only a couple of renames and argument movements.
`StateSpace` now contains `VectorSpace`.
`CreateState` -> `Create`
and StateSpaces and Simulators no longer have qubit size associated with them so we don't have to create new StateSpace and Simulator objects everytime a state vector size changed.

### CODE REVIEW

**Review by zaqqwerty — CHANGES_REQUESTED** [review]
Quick code consistency change, then LGTM :)

File: tensorflow_quantum/core/ops/math_ops/tfq_inner_product.cc [hunk 1]
```diff
@@ -221,15 +219,15 @@ class TfqInnerProductOp : public tensorflow::OpKernel {
       int largest_nq = 1;
       int cur_internal_index;
 
-      State sv = StateSpace(largest_nq, tfq_for).CreateState();
-      State scratch = StateSpace(largest_nq, tfq_for).CreateState();
+      Simulator sim = Simulator(tfq_for);
+      StateSpace ss = StateSpace(tfq_for);
+      auto sv = StateSpace(tfq_for).Create(largest_nq);
```
[the remark this answers is not in the document]
Comment (MichaelBroughton) [reply tensorflow_quantum/core/ops/math_ops/tfq_inner_product.cc]: Fiex.

File: tensorflow_quantum/core/ops/math_ops/tfq_inner_product.cc [hunk 2]
```diff
@@ -155,25 +155,24 @@ class TfqInnerProductOp : public tensorflow::OpKernel {
...
     using StateSpace = Simulator::StateSpace;
-    using State = StateSpace::State;
 
     // Begin simulation.
     int largest_nq = 1;
-    State sv = StateSpace(largest_nq, tfq_for).CreateState();
-    State scratch = StateSpace(largest_nq, tfq_for).CreateState();
+    Simulator sim = Simulator(tfq_for);
+    StateSpace ss = StateSpace(tfq_for);
+    auto sv = StateSpace(tfq_for).Create(largest_nq);
```
[the remark this answers is not in the document]
Comment (MichaelBroughton) [reply tensorflow_quantum/core/ops/math_ops/tfq_inner_product.cc]: Done.

**Review by zaqqwerty — DISMISSED** [review]
LGTM

**Pull request merged**